pkgs/applications/version-management/tortoisehg/default.nix

```diff
@@ -14,40 +14,44 @@ python3Packages.buildPythonApplication rec {
     sha256 = "sha256-Xbvg/FcuX/AL2reWsaM2oaFyLby3+HDCfYtRyswE7DA=";
   };
 
-  # Extension point for when thg's mercurial is lagging behind mainline.
-  tortoiseMercurial = mercurial;
-
+  nativeBuildInputs = [
+    qt5.wrapQtAppsHook
+  ];
   propagatedBuildInputs = with python3Packages; [
-    tortoiseMercurial
-    qscintilla-qt5
+    mercurial
     iniparse
+    # The one from python3Packages
+    qscintilla-qt5
+  ];
+  buildInputs = [
+    # Makes wrapQtAppsHook add these qt libraries to the wrapper search paths
+    qt5.qtwayland
   ];
-  nativeBuildInputs = [ qt5.wrapQtAppsHook ];
 
-  doCheck = true;
+  # In order to spare double wrapping, we use:
+  preFixup = ''
+    makeWrapperArgs+=("''${qtWrapperArgs[@]}")
+  '';
+  # Convenient alias
   postInstall = ''
-    mkdir -p $out/share/doc/tortoisehg
-    cp COPYING.txt $out/share/doc/tortoisehg/Copying.txt
-    # convenient alias
     ln -s $out/bin/thg $out/bin/tortoisehg
-    wrapQtApp $out/bin/thg
   '';
 
+  # In python3Packages.buildPythonApplication doCheck is always true, and we
+  # override it to not run the default unittests
   checkPhase = ''
-    export QT_QPA_PLATFORM=offscreen
-    echo "test: thg smoke test"
-    $out/bin/thg -h > help.txt &
-    sleep 1s
-    if grep "list of commands" help.txt; then
```

Comment:
Is there a reason for removing this test? You've basically reverted my change to a previous state, that was found to be not sufficient in a case of mercurial version mismatch - https://github.com/NixOS/nixpkgs/pull/156146 .

Reply:
> Is there a reason for removing this test? You've basically reverted my change to a previous state, that was found to be not sufficient in a case of mercurial version mismatch - #156146 .

Thanks for commenting @gbtb . Indeed this information was missing, I didn't know about the behavior you described in #156146. I now readded a simpler version of what was there before, with a comment, so that people like me in the future won't be tempted to do the same as I did. For next time, comments are better then commit messages, precisely for these kind of situations!